engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/schemaregistry/confluent/ConfluentSchemaRegistryProvider.scala

```diff
@@ -14,11 +16,17 @@ class ConfluentSchemaRegistryProvider[T: TypeInformation](val schemaRegistryClie
                                                           val deserializationSchemaFactory: KafkaVersionAwareDeserializationSchemaFactory[T],
                                                           val kafkaConfig: KafkaConfig,
                                                           val formatKey: Boolean) extends SchemaRegistryProvider[T] {
+
+  private lazy val schemaRegistryClient = schemaRegistryClientFactory.createSchemaRegistryClient(kafkaConfig)
+
   def recordFormatter(topic: String): Option[RecordFormatter] =
     Some(ConfluentAvroToJsonFormatter(createSchemaRegistryClient, topic, formatKey))
 
   override def createSchemaRegistryClient: ConfluentSchemaRegistryClient =
-    schemaRegistryClientFactory.createSchemaRegistryClient(kafkaConfig)
+    schemaRegistryClient
+
+  override def getSchema(topic: String, version: Option[Int]): Validated[SchemaRegistryError, Schema] =
+    schemaRegistryClient.getSchema(ConfluentUtils.valueSubject(topic), version)
```

Comment:
you change getFreshSchema to getSchema. Is it on purpose?

I also wonder if this is  good place to handle adding "-value". We expose "raw" SchemaRegistryClient as well, so it's possible now to do:

schemaRegistryProvider.schemaRegistryClientFactory.createSchemaRegistry().getSchema
and 
schemaRegistryProvider.getSchema 

and get different results - which is misleading...

Reply:
You're right. I've changed it again ;)